go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/containerd/nerdctl
 
-go 1.17
+go 1.18
```

Comment:
Probably we should still keep this 1.17 for several months

Reply:
Is there any reason?

Reply:
Because users and packagers may take several months to migrate to 1.18

Reply:
If the user is not using the latest version, I think it will be fine. 

In other words, if users need to use the latest version (maybe > 0.18), they should migrate to 1.18.

Reply:
containerd still specifies go 1.17 in go.mod, though CI uses Go 1.18 now

https://github.com/containerd/containerd/blob/ed1a762707cea5055dec47ed4b049159ac9f23a3/go.mod#L3

Reply:
OK, let's be consistent